hudi-bench/src/main/java/org/apache/hudi/bench/dag/nodes/DagNode.java

```diff
@@ -0,0 +1,104 @@
+package org.apache.hudi.bench.dag.nodes;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.hudi.bench.configuration.DeltaConfig.Config;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+/**
+ * Represents a Node in the DAG of operations for a workflow.
+ */
+public abstract class DagNode<O> implements Comparable<DagNode<O>> {
```

Comment:
Since every subclass of `DagNode` has defined a `execute` and we did not use its return value in `DagScheduler`. 
Shall we define an abstract method which signature is:

```
  public abstract void execute(ExecutionContext context) throws Exception;
```

To provide different runtime context information, we can define a context POJO, e.g. `ExecutionContext`.

What do you think?

Reply:
The idea was to have the execute method return it's output so we can actually perform validations at any level in the DAG if needed. 
If you take a look at ValidateNode, that's what it does, it just uses the output of the last execution. Let me think about this a little more.